{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "855fc74c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "pd.options.mode.chained_assignment = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "49066e05",
   "metadata": {},
   "outputs": [],
   "source": [
    "abund_tb=pd.read_csv(\"/Users/hhakimjavadi/Dropbox (UFL)/microbiome_microbiology_CHLA/data/UNITE/BFTMS23071402 (single)OTU (Table).csv\")\n",
    "trim_tbl=pd.read_csv(\"/Users/hhakimjavadi/Dropbox (UFL)/microbiome_microbiology_CHLA/data/Trim_summary.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "beda1018-e73e-4332-91bc-69d247e27fdd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Taxonomy</th>\n",
       "      <th>Combined Abundance</th>\n",
       "      <th>Min</th>\n",
       "      <th>Max</th>\n",
       "      <th>Mean</th>\n",
       "      <th>Median</th>\n",
       "      <th>Std</th>\n",
       "      <th>BFTMS23071402 Abundance</th>\n",
       "      <th>...</th>\n",
       "      <th>BFTMS23071409 Abundance</th>\n",
       "      <th>BFTMS23071410 Abundance</th>\n",
       "      <th>BFTMS23071411 Abundance</th>\n",
       "      <th>BFTMS23071412 Abundance</th>\n",
       "      <th>BFTMS23071413 Abundance</th>\n",
       "      <th>BFTMS23071414 Abundance</th>\n",
       "      <th>BFTMS23071415 Abundance</th>\n",
       "      <th>BFTMS23071416 Abundance</th>\n",
       "      <th>BFTMS23071417 Abundance</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Unidentified; SH489354.07FU_KP012839_reps_sing...</td>\n",
       "      <td>SH489354.07FU_KP012839_reps_singleton</td>\n",
       "      <td>Fungi; Basidiomycota; Agaricomycetes; Agarical...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.1250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.341565</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TCAAAGATTCGATGATTCACGGAATTCTGCAATTCACATTACGTAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Kazachstania_aerobia; SH006357.07FU_AY582126_refs</td>\n",
       "      <td>SH006357.07FU_AY582126_refs</td>\n",
       "      <td>Fungi; Ascomycota; Saccharomycetes; Saccharomy...</td>\n",
       "      <td>26</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1.6250</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.334524</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>TGAAGAACGCAGCGAAATGCGATACGTAATGTGAATTGCAGAATTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Saccharomyces_cerevisiae; SH478399.07FU_GQ2548...</td>\n",
       "      <td>SH478399.07FU_GQ254808_reps_singleton</td>\n",
       "      <td>Fungi; Ascomycota; Saccharomycetes; Saccharomy...</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>1.1250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.247221</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>TGAAGAACGCAGCGAAATGCGATACGTAATGTGAATTGCAGAATTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Kazachstania_hellenica; SH023711.07FU_FJ810522...</td>\n",
       "      <td>SH023711.07FU_FJ810522_refs</td>\n",
       "      <td>Fungi; Ascomycota; Saccharomycetes; Saccharomy...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.1250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.341565</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>TGAAGAACGCAGCGAAATGCGATACGTAATGTGAATTGCAGAATTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Unidentified; SH629179.07FU_KY105124_reps_sing...</td>\n",
       "      <td>SH629179.07FU_KY105124_reps_singleton</td>\n",
       "      <td>Fungi; Ascomycota; Saccharomycetes; Saccharomy...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.1875</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.403113</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TTAAGAACATTGTTCGCCTAGACGCTCTCTTCTTATCGATAACGTT...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  ID  \\\n",
       "0  Unidentified; SH489354.07FU_KP012839_reps_sing...   \n",
       "1  Kazachstania_aerobia; SH006357.07FU_AY582126_refs   \n",
       "2  Saccharomyces_cerevisiae; SH478399.07FU_GQ2548...   \n",
       "3  Kazachstania_hellenica; SH023711.07FU_FJ810522...   \n",
       "4  Unidentified; SH629179.07FU_KY105124_reps_sing...   \n",
       "\n",
       "                                    Name  \\\n",
       "0  SH489354.07FU_KP012839_reps_singleton   \n",
       "1            SH006357.07FU_AY582126_refs   \n",
       "2  SH478399.07FU_GQ254808_reps_singleton   \n",
       "3            SH023711.07FU_FJ810522_refs   \n",
       "4  SH629179.07FU_KY105124_reps_singleton   \n",
       "\n",
       "                                            Taxonomy  Combined Abundance  Min  \\\n",
       "0  Fungi; Basidiomycota; Agaricomycetes; Agarical...                   2    0   \n",
       "1  Fungi; Ascomycota; Saccharomycetes; Saccharomy...                  26    0   \n",
       "2  Fungi; Ascomycota; Saccharomycetes; Saccharomy...                  18    0   \n",
       "3  Fungi; Ascomycota; Saccharomycetes; Saccharomy...                   2    0   \n",
       "4  Fungi; Ascomycota; Saccharomycetes; Saccharomy...                   3    0   \n",
       "\n",
       "   Max    Mean  Median       Std  BFTMS23071402 Abundance  ...  \\\n",
       "0    1  0.1250     0.0  0.341565                        1  ...   \n",
       "1    8  1.6250     1.0  2.334524                        1  ...   \n",
       "2    9  1.1250     0.0  2.247221                        1  ...   \n",
       "3    1  0.1250     0.0  0.341565                        0  ...   \n",
       "4    1  0.1875     0.0  0.403113                        1  ...   \n",
       "\n",
       "   BFTMS23071409 Abundance  BFTMS23071410 Abundance  BFTMS23071411 Abundance  \\\n",
       "0                        0                        0                        0   \n",
       "1                        0                        2                        0   \n",
       "2                        0                        2                        0   \n",
       "3                        0                        0                        0   \n",
       "4                        0                        0                        0   \n",
       "\n",
       "   BFTMS23071412 Abundance  BFTMS23071413 Abundance  BFTMS23071414 Abundance  \\\n",
       "0                        0                        0                        0   \n",
       "1                        0                        0                        2   \n",
       "2                        0                        0                        0   \n",
       "3                        0                        0                        0   \n",
       "4                        0                        1                        0   \n",
       "\n",
       "   BFTMS23071415 Abundance  BFTMS23071416 Abundance  BFTMS23071417 Abundance  \\\n",
       "0                        0                        0                        0   \n",
       "1                        1                        0                        8   \n",
       "2                        1                        1                        9   \n",
       "3                        0                        0                        1   \n",
       "4                        0                        0                        0   \n",
       "\n",
       "                                            Sequence  \n",
       "0  TCAAAGATTCGATGATTCACGGAATTCTGCAATTCACATTACGTAT...  \n",
       "1  TGAAGAACGCAGCGAAATGCGATACGTAATGTGAATTGCAGAATTC...  \n",
       "2  TGAAGAACGCAGCGAAATGCGATACGTAATGTGAATTGCAGAATTC...  \n",
       "3  TGAAGAACGCAGCGAAATGCGATACGTAATGTGAATTGCAGAATTC...  \n",
       "4  TTAAGAACATTGTTCGCCTAGACGCTCTCTTCTTATCGATAACGTT...  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "abund_tb.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "f727b4c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rename the columns by removing the keyword abundance and BFTMS\n",
    "abund_tb.columns = [col.replace(\"Abundance\", '') for col in abund_tb.columns]\n",
    "abund_tb.columns = [col.replace(\" \", '') for col in abund_tb.columns]\n",
    "#abund_tb.columns = [col.replace(\"BFTMS\", '') for col in abund_tb.columns]\n",
    "\n",
    "##add a control column that gets values from the spcified control sample\n",
    "control=\"BFTMS23071417\"\n",
    "#abund_tb[\"control\"]=abund_tb[control]\n",
    "\n",
    "samples=trim_tbl[\"Name\"].astype(str).tolist()\n",
    "#samples.append('control')\n",
    "\n",
    "abund_tb['reference_lib']='UNITE'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "c7eef107-893e-49ee-8f86-4e03b2b51319",
   "metadata": {},
   "outputs": [],
   "source": [
    "## mark species with high risk of false positive detection in a column \"false_pos_prone\"  \n",
    "\n",
    "# first split the ID column into species group name and species ID\n",
    "abund_tb[['species group', 'species id']] = abund_tb['ID'].str.split(';', expand=True)\n",
    "## create a new column \"false_pos_prone\" and mark species with high risk of false positive detection\n",
    "abund_tb['false_pos_prone']='No'\n",
    "\n",
    "\n",
    "def mark_species(df, ID_column, false_pos_tag, species_list):\n",
    "    # Update Column B to True for rows where values in Column A start with any of the strings in the search list\n",
    "    for name in species_list:\n",
    "        df.loc[df[ID_column].str.startswith(name), false_pos_tag] = 'Yes'\n",
    "    return df\n",
    "\n",
    "false_pos_prone_spicies=['000000']\n",
    "abund_tb = mark_species(abund_tb, 'species group', 'false_pos_prone', false_pos_prone_spicies)\n",
    "#abund_tb.loc[abund_tb['species group'].contains(false_pos_prone_spicies), 'false_pos_prone'] = 'Yes'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "0fd7d824",
   "metadata": {},
   "outputs": [],
   "source": [
    "## data normalization and corection for control\n",
    "control=\"BFTMS23071417\"\n",
    "resid_val=0.00 # this value will be added to sample and control to avoid deviding to 0\n",
    "\n",
    "# normalize all abundances based on the total reads for each sample\n",
    "for samp in samples:\n",
    "    ## read the trimmed reads from trim_tbl and get the value based on the sample name\n",
    "    trimmed_reads=trim_tbl[trim_tbl[\"Name\"]==samp][\"Trimmed sequences\"].values[0]\n",
    "    abund_tb[samp+\"_Normalized\"]=abund_tb[samp]*trimmed_reads/1000000\n",
    "\n",
    "## correct normized abundance by deducing control abundance\n",
    "for samp in samples:\n",
    "    abund_tb[samp+\"_Normalized_corrected\"]=abund_tb[samp+\"_Normalized\"]-abund_tb[control+\"_Normalized\"]\n",
    "    ## calculate signal to noize \n",
    "    abund_tb[samp+\"_S:N\"]=(abund_tb[samp+\"_Normalized_corrected\"]+resid_val)/(abund_tb[control+\"_Normalized\"]+resid_val)\n",
    "\n",
    "## ## calculate signal to noize\n",
    "for samp in samples:\n",
    "    abund_tb[samp+\"_S:N\"]=(abund_tb[samp+\"_Normalized_corrected\"]+resid_val)/(abund_tb[control+\"_Normalized\"]+resid_val)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "c47e8c5f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>BFTMS23071405</th>\n",
       "      <th>BFTMS23071417</th>\n",
       "      <th>BFTMS23071405_Normalized</th>\n",
       "      <th>BFTMS23071417_Normalized</th>\n",
       "      <th>BFTMS23071405_Normalized_corrected</th>\n",
       "      <th>BFTMS23071405_S:N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Unidentified; SH489354.07FU_KP012839_reps_sing...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Kazachstania_aerobia; SH006357.07FU_AY582126_refs</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>0.346402</td>\n",
       "      <td>0.805624</td>\n",
       "      <td>-0.459222</td>\n",
       "      <td>-0.570020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Saccharomyces_cerevisiae; SH478399.07FU_GQ2548...</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>0.346402</td>\n",
       "      <td>0.906327</td>\n",
       "      <td>-0.559925</td>\n",
       "      <td>-0.617796</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Kazachstania_hellenica; SH023711.07FU_FJ810522...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.100703</td>\n",
       "      <td>-0.100703</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Unidentified; SH629179.07FU_KY105124_reps_sing...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  ID  BFTMS23071405  \\\n",
       "0  Unidentified; SH489354.07FU_KP012839_reps_sing...              0   \n",
       "1  Kazachstania_aerobia; SH006357.07FU_AY582126_refs              2   \n",
       "2  Saccharomyces_cerevisiae; SH478399.07FU_GQ2548...              2   \n",
       "3  Kazachstania_hellenica; SH023711.07FU_FJ810522...              0   \n",
       "4  Unidentified; SH629179.07FU_KY105124_reps_sing...              0   \n",
       "\n",
       "   BFTMS23071417  BFTMS23071405_Normalized  BFTMS23071417_Normalized  \\\n",
       "0              0                  0.000000                  0.000000   \n",
       "1              8                  0.346402                  0.805624   \n",
       "2              9                  0.346402                  0.906327   \n",
       "3              1                  0.000000                  0.100703   \n",
       "4              0                  0.000000                  0.000000   \n",
       "\n",
       "   BFTMS23071405_Normalized_corrected  BFTMS23071405_S:N  \n",
       "0                            0.000000                NaN  \n",
       "1                           -0.459222          -0.570020  \n",
       "2                           -0.559925          -0.617796  \n",
       "3                           -0.100703          -1.000000  \n",
       "4                            0.000000                NaN  "
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## check one sample and compare it against the results from manual analysis\n",
    "sample='BFTMS23071405'\n",
    "sample_df=abund_tb[['ID', sample, control,sample+'_Normalized', control+'_Normalized',sample+'_Normalized_corrected', sample+'_S:N']]\n",
    "sample_df.sort_values(by=sample+'_Normalized_corrected', ascending=False)\n",
    "sample_df.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "3ecdb1f7-1c96-4a1f-8348-2a75ce433290",
   "metadata": {},
   "outputs": [],
   "source": [
    "## UNITE strict\n",
    "\n",
    "standard_col_names=['ID','Name','Taxonomy','species group', 'species id','reference_lib','false_pos_prone','Combined','Min','Max','Mean','Median','Std', 'sample_abund', 'control_abund','sample_normalized', 'control_normalized','normalized_corrected', 'S:N', 'sample', 'control']\n",
    "select_IDs=pd.DataFrame(columns=standard_col_names)\n",
    "\n",
    "for sample in samples:\n",
    "    select_columns=['ID','Name','Taxonomy','species group', 'species id','reference_lib','false_pos_prone','Combined','Min','Max','Mean','Median','Std', sample, control,sample+'_Normalized', control+'_Normalized',sample+'_Normalized_corrected', sample+'_S:N']\n",
    "    sample_abund=abund_tb[select_columns]\n",
    "    sample_abund['sample']=sample\n",
    "    sample_abund['control']=control\n",
    "    sample_abund.columns=standard_col_names\n",
    "    #sample_abund.rename(columns={sample: \"sample\",control:\"control\", sample+'_Normalized':\"Normalized'}\n",
    "    sample_abund=sample_abund.sort_values(by='normalized_corrected', ascending=False)\n",
    "    sample_abund=sample_abund[0:15]\n",
    "    \n",
    "\n",
    "    select_ID=sample_abund[((sample_abund['false_pos_prone']=='Yes') &\n",
    "                            (sample_abund['sample_abund']>=1000) & \n",
    "                            ((sample_abund['control_abund']==0) | (sample_abund['S:N']>=1000)) &\n",
    "                            (sample_abund['normalized_corrected']>=0)) \n",
    "                            |\n",
    "                            ((sample_abund['false_pos_prone']=='No') &\n",
    "                            (sample_abund['sample_abund']>=10) & \n",
    "                            ((sample_abund['control_abund']==0) | (sample_abund['S:N']>=1000)) &\n",
    "                            (sample_abund['normalized_corrected']>=5))\n",
    "                            ]\n",
    "    \n",
    "    select_IDs=pd.concat([select_IDs, select_ID], axis=0)\n",
    "\n",
    "select_IDs['mode']='UNITE_strict'\n",
    "select_IDs.to_csv('/Users/hhakimjavadi/Dropbox (UFL)/microbiome_microbiology_CHLA/output_tables/UNITE_strict.csv')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "f7b57b4e-5d0a-47e1-a2b8-f7541cc808af",
   "metadata": {},
   "outputs": [],
   "source": [
    "## UNITE moderate\n",
    "\n",
    "standard_col_names=['ID','Name','Taxonomy','species group', 'species id','reference_lib','false_pos_prone','Combined','Min','Max','Mean','Median','Std', 'sample_abund', 'control_abund','sample_normalized', 'control_normalized','normalized_corrected', 'S:N', 'sample', 'control']\n",
    "select_IDs=pd.DataFrame(columns=standard_col_names)\n",
    "\n",
    "for sample in samples:\n",
    "    select_columns=['ID','Name','Taxonomy','species group', 'species id','reference_lib','false_pos_prone','Combined','Min','Max','Mean','Median','Std', sample, control,sample+'_Normalized', control+'_Normalized',sample+'_Normalized_corrected', sample+'_S:N']\n",
    "    sample_abund=abund_tb[select_columns]\n",
    "    sample_abund['sample']=sample\n",
    "    sample_abund['control']=control\n",
    "    sample_abund.columns=standard_col_names\n",
    "    #sample_abund.rename(columns={sample: \"sample\",control:\"control\", sample+'_Normalized':\"Normalized'}\n",
    "    sample_abund=sample_abund.sort_values(by='normalized_corrected', ascending=False)\n",
    "    sample_abund=sample_abund[0:15]\n",
    "    \n",
    "\n",
    "\n",
    "    select_ID=sample_abund[((sample_abund['false_pos_prone']=='Yes') &\n",
    "                            (sample_abund['sample_abund']>=1000) & \n",
    "                            ((sample_abund['control_abund']==0) | (sample_abund['S:N']>=1000)) &\n",
    "                            (sample_abund['normalized_corrected']>=0)) \n",
    "                            |\n",
    "                            ((sample_abund['false_pos_prone']=='No') &\n",
    "                            (sample_abund['sample_abund']>=7) & \n",
    "                            ((sample_abund['control_abund']==0) | (sample_abund['S:N']>=1000)) &\n",
    "                            (sample_abund['normalized_corrected']>=3))\n",
    "                            ]\n",
    "\n",
    "    select_IDs=pd.concat([select_IDs, select_ID], axis=0)\n",
    "\n",
    "select_IDs['mode']='UNITE_moderate'\n",
    "select_IDs.to_csv('/Users/hhakimjavadi/Dropbox (UFL)/microbiome_microbiology_CHLA/output_tables/UNITE_moderate.csv')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "e49a4c61-0abb-4009-ae39-ba052a27cf63",
   "metadata": {},
   "outputs": [],
   "source": [
    "## LSU loose\n",
    "\n",
    "standard_col_names=['ID','Name','Taxonomy','species group', 'species id','reference_lib','false_pos_prone','Combined','Min','Max','Mean','Median','Std', 'sample_abund', 'control_abund','sample_normalized', 'control_normalized','normalized_corrected', 'S:N', 'sample', 'control']\n",
    "select_IDs=pd.DataFrame(columns=standard_col_names)\n",
    "\n",
    "for sample in samples:\n",
    "    select_columns=['ID','Name','Taxonomy','species group', 'species id','reference_lib','false_pos_prone','Combined','Min','Max','Mean','Median','Std', sample, control,sample+'_Normalized', control+'_Normalized',sample+'_Normalized_corrected', sample+'_S:N']\n",
    "    sample_abund=abund_tb[select_columns]\n",
    "    sample_abund['sample']=sample\n",
    "    sample_abund['control']=control\n",
    "    sample_abund.columns=standard_col_names\n",
    "    #sample_abund.rename(columns={sample: \"sample\",control:\"control\", sample+'_Normalized':\"Normalized'}\n",
    "    sample_abund=sample_abund.sort_values(by='normalized_corrected', ascending=False)\n",
    "    sample_abund=sample_abund[0:15]\n",
    "    \n",
    "\n",
    "    select_ID=sample_abund[((sample_abund['false_pos_prone']=='Yes') &\n",
    "                            (sample_abund['sample_abund']>=1000) & \n",
    "                            ((sample_abund['control_abund']==0) | (sample_abund['S:N']>=1000)) &\n",
    "                            (sample_abund['normalized_corrected']>=0)) \n",
    "                            |\n",
    "                            ((sample_abund['false_pos_prone']=='No') &\n",
    "                            (sample_abund['sample_abund']>=4) & \n",
    "                            ((sample_abund['control_abund']==0) | (sample_abund['S:N']>=3)) &\n",
    "                            (sample_abund['normalized_corrected']>=1.5))\n",
    "                            ]\n",
    "\n",
    "    select_IDs=pd.concat([select_IDs, select_ID], axis=0)\n",
    "\n",
    "select_IDs['mode']='UNITE_loose'\n",
    "select_IDs.to_csv('/Users/hhakimjavadi/Dropbox (UFL)/microbiome_microbiology_CHLA/output_tables/UNITE_loose.csv')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8181243a-9534-4b06-b2f9-abe50fa25d20",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "celltoolbar": "Raw Cell Format",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
