{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "855fc74c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "pd.options.mode.chained_assignment = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "49066e05",
   "metadata": {},
   "outputs": [],
   "source": [
    "abund_tb=pd.read_csv(\"/Users/hhakimjavadi/Dropbox (UFL)/microbiome_microbiology_CHLA/data/PanBac/BFTMS23071402 (single)OTU (Table).csv\")\n",
    "trim_tbl=pd.read_csv(\"/Users/hhakimjavadi/Dropbox (UFL)/microbiome_microbiology_CHLA/data/Trim_summary.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "beda1018-e73e-4332-91bc-69d247e27fdd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Taxonomy</th>\n",
       "      <th>Combined Abundance</th>\n",
       "      <th>Min</th>\n",
       "      <th>Max</th>\n",
       "      <th>Mean</th>\n",
       "      <th>Median</th>\n",
       "      <th>Std</th>\n",
       "      <th>BFTMS23071402 Abundance</th>\n",
       "      <th>...</th>\n",
       "      <th>BFTMS23071409 Abundance</th>\n",
       "      <th>BFTMS23071410 Abundance</th>\n",
       "      <th>BFTMS23071411 Abundance</th>\n",
       "      <th>BFTMS23071412 Abundance</th>\n",
       "      <th>BFTMS23071413 Abundance</th>\n",
       "      <th>BFTMS23071414 Abundance</th>\n",
       "      <th>BFTMS23071415 Abundance</th>\n",
       "      <th>BFTMS23071416 Abundance</th>\n",
       "      <th>BFTMS23071417 Abundance</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Staphylococcus epidermidis; mg|Staphylococcus|...</td>\n",
       "      <td>mg|Staphylococcus|epidermidis|-01</td>\n",
       "      <td>Bacteria; ; ; ; ; ; Staphylococcus epidermidis</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>1.1875</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.561738</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TGATTTCTTAAAATACCTTTATGAAAAAGTACATGTTTTGGAAAAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Staphylococcus epidermidis; mg|Staphylococcus|...</td>\n",
       "      <td>mg|Staphylococcus|epidermidis|-02</td>\n",
       "      <td>Bacteria; ; ; ; ; ; Staphylococcus epidermidis</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>2.4375</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.303694</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>TTAATACTTGGTACATCAATTGTTTGAGACATATCTATAAAGTATT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Pseudomonas aeruginosa; mg|Pseudomonas|aerugin...</td>\n",
       "      <td>mg|Pseudomonas|aeruginosa|MPAO1/P1|-01</td>\n",
       "      <td>Bacteria; ; ; ; ; ; Pseudomonas aeruginosa</td>\n",
       "      <td>132</td>\n",
       "      <td>0</td>\n",
       "      <td>122</td>\n",
       "      <td>8.2500</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.416005</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>122</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TGCAGCGGACGCTGGCAGTATTCGTGGATACGCACGCCCTCGCGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Pseudomonas aeruginosa; mg|Pseudomonas|aerugin...</td>\n",
       "      <td>mg|Pseudomonas|aeruginosa|MPAO1/P1|-02</td>\n",
       "      <td>Bacteria; ; ; ; ; ; Pseudomonas aeruginosa</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>1.3750</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.500000</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>CTTCCTCGGCTTCCACTTTTCCCGGCATTTCCCGGCCATCGCCGGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Citrobacter freundii; mg|Citrobacter|freundii|...</td>\n",
       "      <td>mg|Citrobacter|freundii|ATCC-8090--MTCC-1658|</td>\n",
       "      <td>Bacteria; ; ; ; ; ; Citrobacter freundii</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.1250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.341565</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>GCGAAACGCACCGTGTATTGGGGTTGGAAATGGGGGCCGATGATTA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  ID  \\\n",
       "0  Staphylococcus epidermidis; mg|Staphylococcus|...   \n",
       "1  Staphylococcus epidermidis; mg|Staphylococcus|...   \n",
       "2  Pseudomonas aeruginosa; mg|Pseudomonas|aerugin...   \n",
       "3  Pseudomonas aeruginosa; mg|Pseudomonas|aerugin...   \n",
       "4  Citrobacter freundii; mg|Citrobacter|freundii|...   \n",
       "\n",
       "                                            Name  \\\n",
       "0              mg|Staphylococcus|epidermidis|-01   \n",
       "1              mg|Staphylococcus|epidermidis|-02   \n",
       "2         mg|Pseudomonas|aeruginosa|MPAO1/P1|-01   \n",
       "3         mg|Pseudomonas|aeruginosa|MPAO1/P1|-02   \n",
       "4  mg|Citrobacter|freundii|ATCC-8090--MTCC-1658|   \n",
       "\n",
       "                                         Taxonomy  Combined Abundance  Min  \\\n",
       "0  Bacteria; ; ; ; ; ; Staphylococcus epidermidis                  19    0   \n",
       "1  Bacteria; ; ; ; ; ; Staphylococcus epidermidis                  39    0   \n",
       "2      Bacteria; ; ; ; ; ; Pseudomonas aeruginosa                 132    0   \n",
       "3      Bacteria; ; ; ; ; ; Pseudomonas aeruginosa                  22    0   \n",
       "4        Bacteria; ; ; ; ; ; Citrobacter freundii                   2    0   \n",
       "\n",
       "   Max    Mean  Median        Std  BFTMS23071402 Abundance  ...  \\\n",
       "0    7  1.1875     0.0   2.561738                        0  ...   \n",
       "1   18  2.4375     0.0   5.303694                        0  ...   \n",
       "2  122  8.2500     0.0  30.416005                        0  ...   \n",
       "3   22  1.3750     0.0   5.500000                        0  ...   \n",
       "4    1  0.1250     0.0   0.341565                        0  ...   \n",
       "\n",
       "   BFTMS23071409 Abundance  BFTMS23071410 Abundance  BFTMS23071411 Abundance  \\\n",
       "0                        6                        7                        0   \n",
       "1                        2                        2                        0   \n",
       "2                        0                        9                      122   \n",
       "3                        0                        0                       22   \n",
       "4                        0                        0                        0   \n",
       "\n",
       "   BFTMS23071412 Abundance  BFTMS23071413 Abundance  BFTMS23071414 Abundance  \\\n",
       "0                        0                        0                        0   \n",
       "1                        0                        0                        0   \n",
       "2                        1                        0                        0   \n",
       "3                        0                        0                        0   \n",
       "4                        0                        1                        0   \n",
       "\n",
       "   BFTMS23071415 Abundance  BFTMS23071416 Abundance  BFTMS23071417 Abundance  \\\n",
       "0                        0                        0                        0   \n",
       "1                        0                        0                       13   \n",
       "2                        0                        0                        0   \n",
       "3                        0                        0                        0   \n",
       "4                        0                        1                        0   \n",
       "\n",
       "                                            Sequence  \n",
       "0  TGATTTCTTAAAATACCTTTATGAAAAAGTACATGTTTTGGAAAAA...  \n",
       "1  TTAATACTTGGTACATCAATTGTTTGAGACATATCTATAAAGTATT...  \n",
       "2  TGCAGCGGACGCTGGCAGTATTCGTGGATACGCACGCCCTCGCGCA...  \n",
       "3  CTTCCTCGGCTTCCACTTTTCCCGGCATTTCCCGGCCATCGCCGGT...  \n",
       "4  GCGAAACGCACCGTGTATTGGGGTTGGAAATGGGGGCCGATGATTA...  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "abund_tb.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "f727b4c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rename the columns by removing the keyword abundance and BFTMS\n",
    "abund_tb.columns = [col.replace(\"Abundance\", '') for col in abund_tb.columns]\n",
    "abund_tb.columns = [col.replace(\" \", '') for col in abund_tb.columns]\n",
    "#abund_tb.columns = [col.replace(\"BFTMS\", '') for col in abund_tb.columns]\n",
    "\n",
    "##add a control column that gets values from the spcified control sample\n",
    "control=\"BFTMS23071417\"\n",
    "#abund_tb[\"control\"]=abund_tb[control]\n",
    "\n",
    "samples=trim_tbl[\"Name\"].astype(str).tolist()\n",
    "#samples.append('control')\n",
    "\n",
    "abund_tb['reference_lib']='PanBac'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "7f1ac459-ff4e-4ce2-bbdc-fff182fed273",
   "metadata": {},
   "outputs": [],
   "source": [
    "## mark species with high risk of false positive detection in a column \"false_pos_prone\"  \n",
    "\n",
    "# first split the ID column into species group name and species ID\n",
    "abund_tb[['species group', 'species id']] = abund_tb['ID'].str.split(';', expand=True)\n",
    "## create a new column \"false_pos_prone\" and mark species with high risk of false positive detection\n",
    "abund_tb['false_pos_prone']='No'\n",
    "\n",
    "\n",
    "def mark_species(df, ID_column, false_pos_tag, species_list):\n",
    "    # Update Column B to True for rows where values in Column A start with any of the strings in the search list\n",
    "    for name in species_list:\n",
    "        df.loc[df[ID_column].str.startswith(name), false_pos_tag] = 'Yes'\n",
    "    return df\n",
    "\n",
    "false_pos_prone_spicies=['Acinetobacter baumannii', 'Klebsiella oxytoca', 'Enterobacter cloacae',\n",
    "                         'Citrobacter freundii']\n",
    "abund_tb = mark_species(abund_tb, 'species group', 'false_pos_prone', false_pos_prone_spicies)\n",
    "#abund_tb.loc[abund_tb['species group'].isin(false_pos_prone_spicies), 'false_pos_prone'] = 'Yes'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "0fd7d824",
   "metadata": {},
   "outputs": [],
   "source": [
    "## data normalization and corection for control\n",
    "control=\"BFTMS23071417\"\n",
    "resid_val=0.00 # this value will be added to sample and control to avoid deviding to 0\n",
    "\n",
    "# normalize all abundances based on the total reads for each sample\n",
    "for samp in samples:\n",
    "    ## read the trimmed reads from trim_tbl and get the value based on the sample name\n",
    "    trimmed_reads=trim_tbl[trim_tbl[\"Name\"]==samp][\"Trimmed sequences\"].values[0]\n",
    "    abund_tb[samp+\"_Normalized\"]=abund_tb[samp]*trimmed_reads/1000000\n",
    "\n",
    "## correct normized abundance by deducing control abundance\n",
    "for samp in samples:\n",
    "    abund_tb[samp+\"_Normalized_corrected\"]=abund_tb[samp+\"_Normalized\"]-abund_tb[control+\"_Normalized\"]\n",
    "    ## calculate signal to noize \n",
    "    abund_tb[samp+\"_S:N\"]=(abund_tb[samp+\"_Normalized_corrected\"]+resid_val)/(abund_tb[control+\"_Normalized\"]+resid_val)\n",
    "\n",
    "## ## calculate signal to noize\n",
    "for samp in samples:\n",
    "    abund_tb[samp+\"_S:N\"]=(abund_tb[samp+\"_Normalized_corrected\"]+resid_val)/(abund_tb[control+\"_Normalized\"]+resid_val)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "c47e8c5f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>BFTMS23071405</th>\n",
       "      <th>BFTMS23071417</th>\n",
       "      <th>BFTMS23071405_Normalized</th>\n",
       "      <th>BFTMS23071417_Normalized</th>\n",
       "      <th>BFTMS23071405_Normalized_corrected</th>\n",
       "      <th>BFTMS23071405_S:N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Staphylococcus epidermidis; mg|Staphylococcus|...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Staphylococcus epidermidis; mg|Staphylococcus|...</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>0.692804</td>\n",
       "      <td>1.309139</td>\n",
       "      <td>-0.616335</td>\n",
       "      <td>-0.470794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Pseudomonas aeruginosa; mg|Pseudomonas|aerugin...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Pseudomonas aeruginosa; mg|Pseudomonas|aerugin...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Citrobacter freundii; mg|Citrobacter|freundii|...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  ID  BFTMS23071405  \\\n",
       "0  Staphylococcus epidermidis; mg|Staphylococcus|...              0   \n",
       "1  Staphylococcus epidermidis; mg|Staphylococcus|...              4   \n",
       "2  Pseudomonas aeruginosa; mg|Pseudomonas|aerugin...              0   \n",
       "3  Pseudomonas aeruginosa; mg|Pseudomonas|aerugin...              0   \n",
       "4  Citrobacter freundii; mg|Citrobacter|freundii|...              0   \n",
       "\n",
       "   BFTMS23071417  BFTMS23071405_Normalized  BFTMS23071417_Normalized  \\\n",
       "0              0                  0.000000                  0.000000   \n",
       "1             13                  0.692804                  1.309139   \n",
       "2              0                  0.000000                  0.000000   \n",
       "3              0                  0.000000                  0.000000   \n",
       "4              0                  0.000000                  0.000000   \n",
       "\n",
       "   BFTMS23071405_Normalized_corrected  BFTMS23071405_S:N  \n",
       "0                            0.000000                NaN  \n",
       "1                           -0.616335          -0.470794  \n",
       "2                            0.000000                NaN  \n",
       "3                            0.000000                NaN  \n",
       "4                            0.000000                NaN  "
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## check one sample and compare it against the results from manual analysis\n",
    "sample='BFTMS23071405'\n",
    "sample_df=abund_tb[['ID', sample, control,sample+'_Normalized', control+'_Normalized',sample+'_Normalized_corrected', sample+'_S:N']]\n",
    "sample_df.sort_values(by=sample+'_Normalized_corrected', ascending=False)\n",
    "sample_df.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "3ecdb1f7-1c96-4a1f-8348-2a75ce433290",
   "metadata": {},
   "outputs": [],
   "source": [
    "## PanBac strict\n",
    "\n",
    "standard_col_names=['ID','Name','Taxonomy','species group', 'species id','reference_lib','false_pos_prone','Combined','Min','Max','Mean','Median','Std', 'sample_abund', 'control_abund','sample_normalized', 'control_normalized','normalized_corrected', 'S:N', 'sample', 'control']\n",
    "select_IDs=pd.DataFrame(columns=standard_col_names)\n",
    "\n",
    "for sample in samples:\n",
    "    select_columns=['ID','Name','Taxonomy','species group', 'species id','reference_lib','false_pos_prone','Combined','Min','Max','Mean','Median','Std', sample, control,sample+'_Normalized', control+'_Normalized',sample+'_Normalized_corrected', sample+'_S:N']\n",
    "    sample_abund=abund_tb[select_columns]\n",
    "    sample_abund['sample']=sample\n",
    "    sample_abund['control']=control\n",
    "    sample_abund.columns=standard_col_names\n",
    "    #sample_abund.rename(columns={sample: \"sample\",control:\"control\", sample+'_Normalized':\"Normalized'}\n",
    "    sample_abund=sample_abund.sort_values(by='normalized_corrected', ascending=False)\n",
    "    sample_abund=sample_abund[0:15]\n",
    "\n",
    "    select_ID=sample_abund[((sample_abund['false_pos_prone']=='Yes') &\n",
    "                            (sample_abund['sample_abund']>=500) & \n",
    "                            ((sample_abund['control_abund']==0) | (sample_abund['S:N']>=1000)) &\n",
    "                            (sample_abund['normalized_corrected']>=1)) \n",
    "                            |\n",
    "                            ((sample_abund['false_pos_prone']=='No') &\n",
    "                            (sample_abund['sample_abund']>=10) & \n",
    "                            ((sample_abund['control_abund']==0) | (sample_abund['S:N']>=1000)) &\n",
    "                            (sample_abund['normalized_corrected']>=1))\n",
    "                            ]\n",
    "    select_IDs=pd.concat([select_IDs, select_ID], axis=0)\n",
    "\n",
    "select_IDs['mode']='PanBac_strict'\n",
    "select_IDs.to_csv('/Users/hhakimjavadi/Dropbox (UFL)/microbiome_microbiology_CHLA/output_tables/PanBac_strict.csv')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "f7b57b4e-5d0a-47e1-a2b8-f7541cc808af",
   "metadata": {},
   "outputs": [],
   "source": [
    "## PanBac moderate\n",
    "\n",
    "standard_col_names=['ID','Name','Taxonomy','species group', 'species id','reference_lib','false_pos_prone','Combined','Min','Max','Mean','Median','Std', 'sample_abund', 'control_abund','sample_normalized', 'control_normalized','normalized_corrected', 'S:N', 'sample', 'control']\n",
    "select_IDs=pd.DataFrame(columns=standard_col_names)\n",
    "\n",
    "for sample in samples:\n",
    "    select_columns=['ID','Name','Taxonomy','species group', 'species id','reference_lib','false_pos_prone','Combined','Min','Max','Mean','Median','Std', sample, control,sample+'_Normalized', control+'_Normalized',sample+'_Normalized_corrected', sample+'_S:N']\n",
    "    sample_abund=abund_tb[select_columns]\n",
    "    sample_abund['sample']=sample\n",
    "    sample_abund['control']=control\n",
    "    sample_abund.columns=standard_col_names\n",
    "    #sample_abund.rename(columns={sample: \"sample\",control:\"control\", sample+'_Normalized':\"Normalized'}\n",
    "    sample_abund=sample_abund.sort_values(by='normalized_corrected', ascending=False)\n",
    "    sample_abund=sample_abund[0:15]\n",
    "    \n",
    "    select_ID=sample_abund[((sample_abund['false_pos_prone']=='Yes') &\n",
    "                            (sample_abund['sample_abund']>=300) & \n",
    "                            ((sample_abund['control_abund']==0) | (sample_abund['S:N']>=1000)) &\n",
    "                            (sample_abund['normalized_corrected']>=1)) \n",
    "                            |\n",
    "                            ((sample_abund['false_pos_prone']=='No') &\n",
    "                            (sample_abund['sample_abund']>=10) & \n",
    "                            ((sample_abund['control_abund']==0) | (sample_abund['S:N']>=1000)) &\n",
    "                            (sample_abund['normalized_corrected']>=1))\n",
    "                            ]\n",
    "    select_IDs=pd.concat([select_IDs, select_ID], axis=0)\n",
    "\n",
    "select_IDs['mode']='PanBac_moderate'\n",
    "select_IDs.to_csv('/Users/hhakimjavadi/Dropbox (UFL)/microbiome_microbiology_CHLA/output_tables/PanBac_moderate.csv')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "e49a4c61-0abb-4009-ae39-ba052a27cf63",
   "metadata": {},
   "outputs": [],
   "source": [
    "## PanBac loose\n",
    "\n",
    "standard_col_names=['ID','Name','Taxonomy','species group', 'species id','reference_lib','false_pos_prone','Combined','Min','Max','Mean','Median','Std', 'sample_abund', 'control_abund','sample_normalized', 'control_normalized','normalized_corrected', 'S:N', 'sample', 'control']\n",
    "select_IDs=pd.DataFrame(columns=standard_col_names)\n",
    "\n",
    "for sample in samples:\n",
    "    select_columns=['ID','Name','Taxonomy','species group', 'species id','reference_lib','false_pos_prone','Combined','Min','Max','Mean','Median','Std', sample, control,sample+'_Normalized', control+'_Normalized',sample+'_Normalized_corrected', sample+'_S:N']\n",
    "    sample_abund=abund_tb[select_columns]\n",
    "    sample_abund['sample']=sample\n",
    "    sample_abund['control']=control\n",
    "    sample_abund.columns=standard_col_names\n",
    "    #sample_abund.rename(columns={sample: \"sample\",control:\"control\", sample+'_Normalized':\"Normalized'}\n",
    "    sample_abund=sample_abund.sort_values(by='normalized_corrected', ascending=False)\n",
    "    sample_abund=sample_abund[0:15]\n",
    "    \n",
    "    select_ID=sample_abund[((sample_abund['false_pos_prone']=='Yes') &\n",
    "                            (sample_abund['sample_abund']>=50) & \n",
    "                            ((sample_abund['control_abund']==0) | (sample_abund['S:N']>=1000)) &\n",
    "                            (sample_abund['normalized_corrected']>=1)) \n",
    "                            |\n",
    "                            ((sample_abund['false_pos_prone']=='No') &\n",
    "                            (sample_abund['sample_abund']>=5) & \n",
    "                            ((sample_abund['control_abund']==0) | (sample_abund['S:N']>=1000)) &\n",
    "                            (sample_abund['normalized_corrected']>=1))\n",
    "                            ]\n",
    "\n",
    "    select_IDs=pd.concat([select_IDs, select_ID], axis=0)\n",
    "\n",
    "select_IDs['mode']='PanBac_loose'\n",
    "select_IDs.to_csv('/Users/hhakimjavadi/Dropbox (UFL)/microbiome_microbiology_CHLA/output_tables/PanBac_loose.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8181243a-9534-4b06-b2f9-abe50fa25d20",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "500581e7-e99c-40c5-8d1c-8a078a8e36e1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "celltoolbar": "Raw Cell Format",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
